maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultArtifactResolver.java

```
@@ -426,6 +442,14 @@ else if ( local.getFile() != null )
             }
         }
 
+        for ( ArtifactResolverPostProcessor artifactResolverPostProcessor : artifactResolverPostProcessors.values() )
+        {
+            if ( artifactResolverPostProcessor.postProcess( session, results ) )
+            {
+                failures = true;
+            }
```

This is not fail fast, right?

---

No, goal is (just like with existing resolve code) to let all process, so have summary (all errors) reported, not one by one

---

Since all of them are processed, is it worthwhile to document that PP impls shall not rely on any order and/or the result of a previous call?

---

all the instances are getting the "results list" of resolver, there is no order defined or "result" from previous call.... all PP can do is "augment" (alter) the state of result (in list), or not, implementation specific (ie. the PP in this PR when "recording" will not alter anything, just collect the checksums of resolved artifacts, it does not affect the outcome of resolver in any way).